src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -413,3 +413,70 @@ def do_observation(self, event) -> None:
         self.responses = pd.concat(
             [self.responses, pop_included_mothers] + list(pop_included.values())
         )[self.output_columns]
+
+
+class SocialSecurityObserver(BaseObserver):
+    """Class for observing columns relevant to Social Security registry.
+
+    """
+
+    ADDITIONAL_INPUT_COLUMNS = ["alive", "entrance_time", "exit_time", "ssn"]
+    OUTPUT_COLUMNS = [
+        "first_name_id",
+        "middle_name_id",
+        "last_name_id",
+        "date_of_birth",
+        "event_type",
+        "event_date",
+    ]
+
+    def __repr__(self):
+        return f"SocialSecurityObserver()"
+
+    @property
+    def name(self):
+        return f"social_security_observer"
+
+    @property
+    def output_filename(self):
+        return f"social_security.hdf"
+
+    @property
+    def input_columns(self):
+        return self.DEFAULT_INPUT_COLUMNS + self.ADDITIONAL_INPUT_COLUMNS
+
+    @property
+    def output_columns(self):
+        return self.OUTPUT_COLUMNS
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.clock = builder.time.clock()
+        self.end_time = dt.datetime(**builder.configuration.time.end)
+
+    def to_observe(self, event: Event) -> bool:
+        """Only observe if this is the final time step of the sim"""
+        return self.clock() <= self.end_time < event.time
```

Comment:
I'm not sure about the edge case here, but I think it's safer to do this since I don't think a new time-step would be triggered if `self.end_time == event.time` and therefore we might not record anything. 
```suggestion
        return self.clock() < self.end_time <= event.time
```
If we are confident in vivarium's behavior around that edge case, we could get rid of the `self.clock()` check entirely.

Reply:
I tested this by changing the end time to be a time step, which confirms that @rmudambi 's approach to the edge case achieves the intended result.